portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,33 +20,54 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
+import com.google.sps.data.Message;
 import java.util.ArrayList;
 
 /** Servlet that returns comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  public static final Gson GSON = new Gson();
+  private static final Gson GSON = new Gson();
+  private ArrayList<Message> messages = new ArrayList<>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Create an ArrayList containing hard-coded messages
-    ArrayList<String> messages = new ArrayList<>();
-    messages.add(new String("How are you?"));
-    messages.add(new String("Welcome to my site!"));
-    messages.add(new String("Have a good day!"));
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJsonUsingGson(messages));
+  }
 
-    // Convert the ArrayList to JSON
-    String json = convertToJsonUsingGson(messages);
+  @Override
+  public synchronized void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
How about make the synchronized scope smaller?
Just the messages.add need to be synchronized.